crates/builder/src/tasks/submit.rs

```diff
@@ -53,7 +53,7 @@ impl SubmitTask {
     /// blocks. Seems fine lol.
     async fn get_next_sequence(&self) -> eyre::Result<u64> {
         self.zenith
-            .nextSequence(U256::from(self.config.ru_chain_id))
+            // .nextSequence(U256::from(self.config.ru_chain_id))
```

Comment:
whole function can be deleted. `get_confirm_by` can be deleted too. Instead, we need `async fn next_host_block_height(&self) -> eyre::Result<u64>` which we use to populate the `hostBlockNumber` in `Zenith::BlockHeader`